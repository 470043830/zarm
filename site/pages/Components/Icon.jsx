import Markdown from '@site/components/Markdown';

export default class Page extends Markdown {
  document = () => {
    return {
      document: require('@/components/icon/demo.md'),
      className: 'icon-page',
    };
  }
}
