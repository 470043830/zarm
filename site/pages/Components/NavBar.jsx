import Markdown from '@site/components/Markdown';

export default class Page extends Markdown {
  document = () => {
    return {
      document: require('@/components/nav-bar/demo.md'),
      className: 'nav-bar-page',
    };
  }
}
