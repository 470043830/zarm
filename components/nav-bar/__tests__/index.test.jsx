import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavBar from '../index';
import Icon from '../../icon';

describe('NavBar', () => {
  const props = {};

  it('renders correctly', () => {
    const wrapper = render(
      <NavBar {...props} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with title', () => {
    props.title = '测试标题';
    const wrapper = render(
      <NavBar {...props} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with only left prop', () => {
    props.left = true;
    const wrapper = render(
      <NavBar {...props} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with left content', () => {
    props.left = <Icon type="arrow-left" />;
    const wrapper = render(
      <NavBar {...props} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with right content', () => {
    props.right = <Icon type="arrow-right" />;
    const wrapper = render(
      <NavBar {...props} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with multiple right icons', () => {
    props.right = <div><Icon type="arrow-right" /><Icon type="info-round" /></div>;
    const wrapper = render(
      <NavBar {...props} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
