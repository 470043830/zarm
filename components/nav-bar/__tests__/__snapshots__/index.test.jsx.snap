// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<div
  class="za-nav-bar"
>
  <div
    class="za-nav-bar-title"
  />
  <div
    class="za-nav-bar-right za-nav-bar-icon"
  />
</div>
`;

exports[`NavBar renders correctly with left content 1`] = `
<div
  class="za-nav-bar"
>
  <div
    class="za-nav-bar-left za-nav-bar-icon"
  >
    <i
      class="za-icon za-icon-arrow-left"
    />
  </div>
  <div
    class="za-nav-bar-title"
  >
    测试标题
  </div>
  <div
    class="za-nav-bar-right za-nav-bar-icon"
  />
</div>
`;

exports[`NavBar renders correctly with multiple right icons 1`] = `
<div
  class="za-nav-bar"
>
  <div
    class="za-nav-bar-left za-nav-bar-icon"
  >
    <i
      class="za-icon za-icon-arrow-left"
    />
  </div>
  <div
    class="za-nav-bar-title"
  >
    测试标题
  </div>
  <div
    class="za-nav-bar-right za-nav-bar-icon"
  >
    <div>
      <i
        class="za-icon za-icon-arrow-right"
      />
      <i
        class="za-icon za-icon-info-round"
      />
    </div>
  </div>
</div>
`;

exports[`NavBar renders correctly with only left prop 1`] = `
<div
  class="za-nav-bar"
>
  <div
    class="za-nav-bar-left za-nav-bar-icon"
  >
    <i
      class="za-icon za-icon-arrow-left"
    />
  </div>
  <div
    class="za-nav-bar-title"
  >
    测试标题
  </div>
  <div
    class="za-nav-bar-right za-nav-bar-icon"
  />
</div>
`;

exports[`NavBar renders correctly with right content 1`] = `
<div
  class="za-nav-bar"
>
  <div
    class="za-nav-bar-left za-nav-bar-icon"
  >
    <i
      class="za-icon za-icon-arrow-left"
    />
  </div>
  <div
    class="za-nav-bar-title"
  >
    测试标题
  </div>
  <div
    class="za-nav-bar-right za-nav-bar-icon"
  >
    <i
      class="za-icon za-icon-arrow-right"
    />
  </div>
</div>
`;

exports[`NavBar renders correctly with title 1`] = `
<div
  class="za-nav-bar"
>
  <div
    class="za-nav-bar-title"
  >
    测试标题
  </div>
  <div
    class="za-nav-bar-right za-nav-bar-icon"
  />
</div>
`;
